.env.sample

```diff
@@ -1,3 +1,6 @@
+# Toggle for server configuration in development when using Docker. 
+export DEV_DOCKER=1
```

Comment:
I don't think this needs to be added to the sample file, as this shouldn't be enabled for native local development. Shouldn't this be added directly to the `docker-compose.yml` file instead?

Reply:
@jtgeibel - my thinking here was that since the `.env` file is read by the dockerized environment, it is a good place to add the toggle, instead of the `docker-compose.yml`, which _could_ be used for hosting. If it is used while hosted, then we've exposed the port too widely. 

I updated my branch using the flag directly within `docker-compose.yml` at first, and I don't mind reverting it. Just figured it would be more flexible this way.

Reply:
@nilslice I'm not sure I follow. *If* `docker-compose.yml` is used for deployment, then wouldn't this environment variable also need to be set for that to work?

At the least, we should drop this from `.env.sample` and mention it to `.env` only under the docker section of `CONTRIBUTING.md`. That is the only scenario where we want to recommend using this variable.

Reply:
I've taken the liberty to rebase the branch and perform the change to the `docker-compose.yml` file, since I was considering working on https://github.com/rust-lang/crates.io/issues/2054, but then found this PR here 😅 

should be good to go from my side 👍